src/PowerShellEditorServices/Services/PowerShell/Utility/CommandHelpers.cs

```diff
@@ -168,5 +172,37 @@ internal static class CommandHelpers
 
             return synopsisString;
         }
+
+        /// <summary>
+        /// Gets all aliases found in the runspace
+        /// </summary>
+        /// <param name="executionService"></param>
+        public static async Task<(Dictionary<string, List<string>>, Dictionary<string, string>)> GetAliasesAsync(IInternalPowerShellExecutionService executionService)
+        {
+            Validate.IsNotNull(nameof(executionService), executionService);
+
+            IEnumerable<CommandInfo> aliases = await executionService.ExecuteDelegateAsync<IEnumerable<CommandInfo>>(
+                nameof(GetAliasesAsync),
+                Execution.ExecutionOptions.Default,
+                (pwsh, _) =>
+                {
+                    CommandInvocationIntrinsics invokeCommand = pwsh.Runspace.SessionStateProxy.InvokeCommand;
+                    return invokeCommand.GetCommands("*", CommandTypes.Alias, nameIsPattern: true);
+                },
+                CancellationToken.None).ConfigureAwait(false);
+
+            foreach (AliasInfo aliasInfo in aliases)
+            {
+                s_cmdletToAliasCache.AddOrUpdate(
+                    aliasInfo.Definition,
+                    (_) => new List<string> { aliasInfo.Name},
+                    (_, v) => { v.Add(aliasInfo.Name); return v; });
```

Comment:
This would beat the purpose, as these 2 lambdas are closures and thus a new instance will be created for every `aliasInfo` object, so it will end up with more allocations.

The overload I suggested to use will generate static delegates can are reused for all the calls to `AddOrUpdate` here, and thus could reduce allocations.

Reply:
The suggested code didn't compile...what's the fix for it?

Reply:
I see [this](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-6.0#system-collections-concurrent-concurrentdictionary-2-addorupdate-1(-0-system-func((-0-0-1))-system-func((-0-1-0-1))-0)) let me see if I can get it right.

Reply:
My suggestion is pseudo code :) I didn't try compiling it. I guess the following would work
```
s_cmdletToAliasCache.AddOrUpdate(
    aliasInfo.Definition,
    (_, arg) => new List<string> { arg },
    (_, v, arg) => { v.Add(arg); return v; },
    aliasInfo.Name);
```

Reply:
I'm getting this:

> No overload for method 'AddOrUpdate' takes 4 arguments

Which is why I made the change I did instead. But I agree the documentation says this overload exists and takes four arguments, and we're using .NET 6 here.

Reply:
If I write it like this:

```c#
                s_cmdletToAliasCache.AddOrUpdate<string>(
                    aliasInfo.Definition,
                    (_, arg) => new List<string> { arg },
                    (_, v, arg) => { v.Add(arg); return v; },
                    aliasInfo.Name);
```

Which I think is right, I get:

> The non-generic method 'ConcurrentDictionary<string, List<string>>.AddOrUpdate(string, List<string>, Func<string, List<string>, List<string>>)' cannot be used with type arguments

Do I have a typo somewhere or a misunderstanding?

Reply:
That's weird. The following works fine in a `net6.0` project:
```c#
    class Program
    {
        static void Main(string[] args)
        {
            ConcurrentDictionary<string, List<string>> cmdletToAliasCache = new(StringComparer.CurrentCultureIgnoreCase);
            string key = "key";
            string value = "value";

            cmdletToAliasCache.AddOrUpdate(
                key,
                (_, arg) => new List<string> { arg },
                (_, v, arg) => { v.Add(arg); return v; },
                value);

            cmdletToAliasCache.TryGetValue(key, out var myValue);
            Console.WriteLine(myValue[0]);
        }
    }
```
![image](https://user-images.githubusercontent.com/127450/148621575-35e6ac5b-4411-4057-96a9-a532d1654dd9.png)

Reply:
This overload only exists in .NET Framework.

Reply:
I wonder if it's because our target framework is `netstandard2.0`.

Reply:
<img width="649" alt="Screen Shot 2022-01-07 at 5 13 13 PM" src="https://user-images.githubusercontent.com/2226434/148625605-75b6d502-41a7-44ee-ad6a-cf46ba32ff9d.png">

Reply:
I see, so PSES is targeting netstandard2.0?
In that case, please keep your original code unchanged.

Reply:
> we're using .NET 6 here

Just so I'm clear :) we are using .NET 6 SDK but the build targets netstandard 2.0, is that right?

Reply:
@daxian-dbw Since we're using `netstandard2.0` and `net461` in order to continue to support Windows PowerShell 5.1, I can't use this overload 😢

Reply:
That's OK. It's an optimization change anyway 😄

Reply:
I've added a TODO to improve this with the better delegate if/when we move to .NET 4.7.2 and/or .NET Standards 2.1 (so really, the day we drop PowerShell 5.1 support).